state/src/impls/shard_level.rs

```diff
@@ -511,7 +511,7 @@ impl<B: Backend + ShardBackend> ShardLevelState<B> {
             .fold(true, |acc, ref asset| acc && (None != asset_scheme.pool().iter().position(|ref x| x == asset)))
             || pool.len() != asset_scheme.pool().len()
         {
-            return Err(TransactionError::InvalidInput.into())
```

Comment:
Don't you think the order of commit is weird?
It must before `209c8ac0a466dbf9664793d7ff870e292e0ae25a` and that commit must use `InvalidOutput` error.

Reply:
I reordered it.